records_mover/creds/base_creds.py

```diff
@@ -137,23 +141,69 @@ def default_gcs_client(self) -> Optional['google.cloud.storage.Client']:
             return self.__default_gcs_client
         try:
             import google.cloud.storage  # noqa
-        except ModuleNotFoundError:
-            logger.debug("google.cloud.storage not installed",
-                         exc_info=True)
+        except ModuleNotFoundError as e:
+            logger.warning(f"google.cloud.storage not installed: {e}")
+            logger.debug("Details:", exc_info=True)
             self.__default_gcs_client = None
             return self.__default_gcs_client
 
         try:
-            self.__default_gcs_client = google.cloud.storage.Client(credentials=gcs_creds)
+            other_args = {}
+            #
+            # For mvrec especially, Records Mover on the command line
+            # tends to use "application default creds", which act like
+            # service user creds but are retrieved by browser oauth.
+            #
+            # To create a google.cloud.storage.Client(), we need a
+            # project - projects in GCP are kind of like accounts in
+            # AWS, but a little more convenient to flip through and
+            # less directly tied to billing.
+            #
+            # So it turns out that `google.cloud.storage.Client()`,
+            # when reading "application default creds", requires a GCP
+            # "project" to be passed in.  Nothing you do on the
+            # command line will make it guess the right one short of
+            # (I'm not kidding) logging in again using a different
+            # method that doesn't get useful creds here, and *then*
+            # setting the project in config.
+            #
+            # So the other only option than passing this project arg
+            # in is:
+            #
+            #   # note the next --project arg is ignored for this purpose
+            #   gcloud auth application-default login --project my-project-here
+            #   gcloud auth login
+            #   # now we can set a gcloud config and have it read:
+            #   gcloud config set project my-project-here
+            #
+            # See
+            # https://github.com/bluelabsio/records-mover/issues/119
+            # for a way we can improve this.
```

Comment:
Hmmm, this feels like a lot of information for a code comment.... Not sure if there is somewhere else to also add this? Or a better place for this overall discussion of project structure in BQ?

Reply:
Right on.  I pushed more of the context into the GitHub issue.